### PR TITLE
Add "sync" command

### PR DESCRIPTION
**Pull request by z4yx**
Related issue: #560 

This PR added a `sync` sub command, which allows user to sync one or more packages .

```console
usage: bandersnatch sync [-h] package [package ...]

positional arguments:
  package     The name of package to sync

optional arguments:
  -h, --help  show this help message and exit
```

After a synchronization, bandersnatch refreshes global index files, while keeps status(serial) file intact. 

Would you please review the PR. If this implementation meets the requirement, I'll add some tests then.

### CODE REVIEW

File: src/bandersnatch/main.py [hunk 1]
```diff
@@ -93,6 +93,16 @@ def _verify_parser(subparsers: argparse._SubParsersAction) -> None:
     v.set_defaults(op="verify")
 
 
+def _sync_parser(subparsers: argparse._SubParsersAction) -> None:
+    m = subparsers.add_parser(
+        "sync", help="Synchronize specific packages with the PyPI master server.",
+    )
+    m.add_argument(
+        "package", nargs="+", help="The name of package to sync",
```
Comment (cooperlees) [inline src/bandersnatch/main.py:101]: NIT: Can we plural this since it’s nargs + (packages)

File: src/bandersnatch/main.py [hunk 2]
```diff
@@ -103,6 +113,8 @@ async def async_main(args: argparse.Namespace, config: ConfigParser) -> int:
             return await bandersnatch.delete.delete_packages(config, args, master)
     elif args.op.lower() == "verify":
         return await bandersnatch.verify.metadata_verify(config, args)
+    elif args.op.lower() == "sync":
+        return await bandersnatch.mirror.mirror(config, args.package)
```
Comment (cooperlees) [inline src/bandersnatch/main.py:117]:
```suggestion
        return await bandersnatch.mirror.mirror(config, args.packages)
```

File: src/bandersnatch/mirror.py [hunk 3]
```diff
@@ -112,17 +113,27 @@ def webdir(self) -> Path:
...
-        self.wrapup_successful_sync()
+        if specific_packages is None:
+            # Changelog-based synchronization
+            await self.determine_packages_to_sync()
+            await self.sync_packages()
+            self.sync_index_page()
+            self.wrapup_successful_sync()
+        else:
+            self.packages_to_sync = {
+                utils.bandersnatch_safe_name(name): 1 for name in specific_packages
```
Comment (cooperlees) [inline src/bandersnatch/mirror.py:133]: What does 1 mean here? Should we make a variable to help explain that? 
```suggestion
                utils.bandersnatch_safe_name(name): LOW_SERIAL for name in specific_packages
```

**Comment by z4yx** [comment]
Thank you for pointing out the coding style issues. I did some changes.

File: src/bandersnatch/mirror.py [hunk 4]
```diff
@@ -112,17 +113,32 @@ def webdir(self) -> Path:
...
+            # Changelog-based synchronization
+            await self.determine_packages_to_sync()
+            await self.sync_packages()
+            self.sync_index_page()
+            self.wrapup_successful_sync()
+        else:
+            # Synchronize specific packages. This method doesn't update the self.statusfile
+
+            # Pass serial number 0 to bypass the stale serial check in Package class
+            SERIAL_DONT_CARE = 0
```
Comment (cooperlees) [inline src/bandersnatch/mirror.py:135]: LoL ... Still clearer than a random int! Haha

**Comment by z4yx** [comment]
Yes, I'm going to add tests. Please keep the PR open awhile.

File: src/bandersnatch/mirror.py [hunk 5]
```diff
@@ -112,17 +113,32 @@ def webdir(self) -> Path:
...
-        self.sync_index_page()
-        self.wrapup_successful_sync()
+        if specific_packages is None:
+            # Changelog-based synchronization
+            await self.determine_packages_to_sync()
+            await self.sync_packages()
+            self.sync_index_page()
+            self.wrapup_successful_sync()
+        else:
+            # Synchronize specific packages. This method doesn't update the self.statusfile
```
Comment (ichard26) [inline src/bandersnatch/mirror.py:132]:
```suggestion
            # Synchronize specific packages. This method doesn't update the
            # self.statusfile
```

The CI complained that the line is too long. 
```console
./src/bandersnatch/mirror.py:132:89: E501 line too long (91 > 88 characters)
```

By any chance are the pre-commit hooks not running before your commits? This should be something that is caught by the Flake8 pre-commit hook. No worries though, it's a trival thing, none of us are immune from making mistakes :)

Comment (z4yx) [reply src/bandersnatch/mirror.py:132]: I didn't get notification from pre-commit hooks. I'll check if it's installed correctly.

Comment (cooperlees) [reply src/bandersnatch/mirror.py:132]: CI should definitely be running pre-commit via tox. Please run tox locally too if you can. Should catch all this.

**Comment by z4yx** [comment]
I've added the test. A trivial one :)

**Pull request merged**